pkg/util/modules/modules.go

```diff
@@ -23,19 +27,29 @@ type Manager struct {
 	modules map[string]*module
 }
 
+// PrivateModule is an option for `RegisterModule` that marks module as private. Modules are public by default.
+func PrivateModule(m *module) {
+	m.public = false
+}
```

Comment:
`PrivateModule` vs `public` flag seems inconsistent. I would not use "private"/"public" terminology at all, just "internal" or "not internal": `InternalModule` option, `m.internal`, `IsInternalModule` method. WDYT?

It's a nit pick, and we can do this in separate PR if you prefer.

Reply:
While trying to rename public -> internal, I got stuck on what should I change method name `PublicModuleNames` method to; `NotInternalModuleName` looks wrong, `ExternalModuleName` sounds ok, but we will have to have code like `! module.internal`, which is not much different than `func PrivateModule(m *module) {m.public = false}`. Then I realized the problem is probably not the terms public/private v.s. internal/not internal, the problem is using boolean to represent the public private concept. Seeing `public = false`, `internal = false`, and `!internal` incurs mental cost for the reader. 

So I came up with using the name "visibility" (I think I borrowed this term from OOP) and the value of it would be descriptive enum values. After changing the name I do feel the reading the code flows better. Let me know what you think, I am curious.